forms/NumericField.php

```diff
@@ -9,6 +9,50 @@
  */
 class NumericField extends TextField {
 
+	public function setValue($value, $data = array()) {
+		require_once "Zend/Locale/Format.php";
+
+		// If passing in a non-string number, or a value
+		// directly from a dataobject then localise this number
+		if ((is_numeric($value) && !is_string($value)) || 
+			($value && $data instanceof DataObject)
+		){
+			$locale = new Zend_Locale(i18n::get_locale());
```

Comment:
I would prefer to follow the `DateField` precedent here and use a `setLocale()` that defaults to `i18n::get_locale()`, make the field a bit more versatile.